client/src/models/execution_layer/model_registry.py

```
@@ -0,0 +1,656 @@
+                target_path = model_path / filename
+
+                if target_path.exists():
+                    continue
+
+                logger.info(f"Fetching external file: {filename} -> {src}")
+                if isinstance(src, str) and src.startswith(("http://", "https://")):
+                    with requests.get(src, stream=True, timeout=timeout) as r:
+                        r.raise_for_status()
+                        with open(target_path, "wb") as f:
+                            for chunk in r.iter_content(chunk_size=1024 * 1024):
+                                logger.debug(f"...")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)